v2/internal/testcommon/samples_tester.go

```diff
@@ -228,3 +235,43 @@ func IsSampleExcluded(path string, exclusions []string) bool {
 	}
 	return false
 }
+
+func (t *SamplesTester) findAndSetARMIDReferences(obj genruntime.ARMMetaObject) (genruntime.ARMMetaObject, error) {
+	resourceRefType := reflect.TypeOf(genruntime.ResourceReference{})
+	visitor := reflecthelpers.NewReflectVisitor()
+	visitor.VisitStruct = func(this *reflecthelpers.ReflectVisitor, it reflect.Value, ctx interface{}) error {
+
+		if it.Type() == resourceRefType {
```

Comment:
Suggest inverting the sense of this **if**, turning it into a proper guard clause.